bolt-modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -75,7 +75,13 @@ def run_task_with_description(task_name, targets, description, task_args = nil)
       task = Bolt::Task.new(name: task_name, files: [{ 'name' => '', 'path' => '' }])
     else
       # TODO: use the compiler injection once PUP-8237 lands
-      task_signature = Puppet::Pal::ScriptCompiler.new(closure_scope.compiler).task_signature(task_name)
+      # Avaliable with Puppet.lookup https://github.com/puppetlabs/puppet/pull/6440/files
+      script_compiler = Puppet.lookup(:pal_script_compiler)
```

Comment:
According to https://github.com/puppetlabs/puppet/pull/6440/files this should be available. This was the PR referenced in the `TODO`. However I was not able to get it to work with something like:
```
              Puppet::Functions.create_function(:'a::myscriptcompilerfunc', Puppet::Functions::InternalFunction) do
                dispatch :myscriptcompilerfunc do
                  script_compiler_param
                  param 'String',:name
                end
                def myscriptcompilerfunc(script_compiler, name)
                  script_compiler.is_a?(Puppet::Pal::ScriptCompiler) ? name : 'no go'
                end
              end
```
Is this something we want to do? It would require some alternate mocking in the spec tests where we use the lookup.

Reply:
I'm gong to wait on this. If the parameter was injected and we did not need to lookup I would do it, but it looks like maybe it is only injected for internal puppet functions?